app/ami.py

```diff
@@ -198,5 +204,15 @@ def estimate_monthly_ami_price(ami_type: str, block_device_mappings: list, ami_n
                 total_cost += estimate_monthly_snapshot_price(snapshot_type, snapshot_size)
     else:
         print(f"WARNING: {ami_name} is a {ami_type} type AMI with the following block_device_mappings: "
-              "{block_device_mappings}")
+              f"{block_device_mappings}")
     return total_cost
+
+
+def is_ami_registered(ami_id: str, ) -> bool:
+    # Use global variable to make API call once instead of every time the method is called since call is costly
+    global ORG_AMIS
+    if not ORG_AMIS:
+        ec2_client = boto3.client('ec2')
+        ORG_AMIS = ec2_client.describe_images(Owners=['self'])
+    # Iterate through each image owned by organization to check if ami id exists, if it does it is registered
+    return True if [i['ImageId'] for i in ORG_AMIS['Images'] if i['ImageId'] == ami_id] else False
```

Comment:
Instead of loading a list of images, I think you could request just `ami_id` to load a single image. If it succeeds, then it must be registered. If boto3 fails to load it then it probably doesn't exist. I think that would be efficient enough that a global variable wouldn't be needed

Reply:
Done.

I ended up using 2 exceptions for the try/except  attempting to find a single AMI with boto3 because typically, if an ami does not exist, `ec2.Image(ami_id).name` throws a `botocore.exceptions.ClientError`, but I encountered a case where an `AttributeError` is thrown that reads: `AttributeError: 'NoneType' object has no attribute 'get'`

After doing a google search, I discovered a bug with an open ticket [here](https://github.com/boto/boto3/issues/925) on the boto3 project’s GitHub page describing how None type is returned for deregistered images, so I included the `AttributeError` as well to cover those cases.